main.tf

```diff
@@ -121,9 +123,9 @@ locals {
       role             = host.tags["Role"]
       privateInterface = "ens5"
     }
-  ]
-  windows_workers = [
-    for host in module.windows_workers.machines : {
+  ] : []
+  windows_workers = var.windows_worker_count > 0 ? [
```

Comment:
is this detection logic necessary?  won't the module.windows_workers list just be an empty list?

Reply:
all the workers/managers/msr need to be initialized to something because we are using them later to concat all the nodes in:
`      hosts = concat(local.managers, local.workers, local.msr, local.windows_workers)
`

Reply:
I just ran it as suggested I get the following error
`╷
│ Error: Invalid index
│ 
│   on main.tf line 128, in locals:
│  128:     for host in module.windows_workers[0].machines : {
│     ├────────────────
│     │ module.windows_workers is empty tuple
│ 
`
There is nothing to stop TF from trying to access the module at index [0]